adage-front/src/app/components/OffersSearch/Offers/PrebookingButton/PrebookingButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { NotificationType, Notification,
+    NotificationComponent, } from 'app/components/Layout/Notification/Notification';
+import React, { useCallback, useState } from 'react';
+import { preBookStock } from 'repository/pcapi/pcapi';
+import { StockType } from 'utils/types';
+import './PrebookingButton.scss';
+
+const PrebookingButton = ({className, stock, canPrebookOffers}: {className?: string, stock: StockType, canPrebookOffers: boolean}) => {
+    const [disableButton, setDisableButton] = useState(false)
+  const [notification, setNotification] = useState<Notification | null>(null)
+  
+  const preBookCurrentStock = useCallback(() => {
+    setDisableButton(true)
+    return preBookStock(stock.id)
+      .then(() =>
+        setNotification(
+          new Notification(
+            NotificationType.success,
+            "Votre préréservation a été effectuée avec succès."
+          )
+        )
+      )
+      .catch(() =>
+        setNotification(
+          new Notification(
+            NotificationType.error,
+            "Impossible de préréserver cette offre.\nVeuillez contacter le support"
+          )
+        )
+      )
+  }, [stock.id])
+  
+    return (
+        <>
+            {canPrebookOffers && <button disabled={disableButton} className={`prebooking-button ${className}`} onClick={preBookCurrentStock}>pré-réserver</button>}
```

Comment:
Si jamais le wording est changé:

```suggestion
            {canPrebookOffers && <button disabled={disableButton} className={`prebooking-button ${className}`} onClick={preBookCurrentStock}>préréserver</button>}
```